BuildMasterAutomation/Functions/Set-BMVariable.ps1

```diff
@@ -98,6 +110,12 @@ function Set-BMVariable
     Set-StrictMode -Version 'Latest'
     Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
+    $Value = ConvertTo-BMOtterScriptExpression -Value $Value
+    if (-not $Value)

```

Comment:
I don't think we want to do this, otherwise we'll never be able to set a BuildMaster variable to an empty vector or map, i.e. `Set-BMVariable -Value @()` won't do anything.

Just set the variable to whatever comes back from `ConvertTo-BMOtterScriptExpression`.

Reply:
Oh! I was trying to do something like `if ($Value -eq $null)` and forgot that in PowerShell world `-not ''` is true...